pkg/util/provider/machinecontroller/userdata.go

```diff
@@ -29,19 +14,27 @@ import (
 	"strings"
 	"time"
 
-	"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	bootstraptokenapi "k8s.io/cluster-bootstrap/token/api"
 	bootstraptokenutil "k8s.io/cluster-bootstrap/token/util"
 	"k8s.io/klog/v2"
+
+	"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
 )
 
-const placeholder = "<<BOOTSTRAP_TOKEN>>"
+const (
+	bootstrapTokenPlaceholder = "<<BOOTSTRAP_TOKEN>>"
+	machineNamePlaceholder    = "<<MACHINE_NAME>>"
+)
 
-// urlEncodedPlaceholder is a placeholder that can for instance occur in ignition userdata format
-var urlEncodedPlaceholder = url.QueryEscape(placeholder)
+var (
+	// urlEncodedBootstrapTokenPlaceholder is a bootstrapTokenPlaceholder that can for instance occur in ignition userdata format
+	urlEncodedBootstrapTokenPlaceholder = url.QueryEscape(bootstrapTokenPlaceholder)
+	// urlEncodedMachineNamePlaceholder is a machineNamePlaceholder that can for instance occur in ignition userdata format
+	urlEncodedMachineNamePlaceholder = url.QueryEscape(machineNamePlaceholder)
+)
 
 func (c *controller) addBootstrapTokenToUserData(ctx context.Context, machine *v1alpha1.Machine, secret *corev1.Secret) error {
 	var (
```

Comment:
we really need to introduce a unit test for this method. Sorry that we don't have one at the moment.

Reply:
I added some tests. In general, it motivates people to add tests for new functions if there are already tests for existing functions 😉